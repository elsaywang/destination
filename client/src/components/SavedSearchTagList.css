:local(.noSavedSearches) {
    line-height: 30px;
    margin-right: 10px;
}
