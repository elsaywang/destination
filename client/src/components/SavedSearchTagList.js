import React from 'react';
import PropTypes from 'prop-types';
import { TagList, Tag } from '@react/react-spectrum/TagList';

function SavedSearchTagList({ list, onSavedSearchClick }) {
    const boundClick = search => onSavedSearchClick.bind(this, search);
    const renderTag = search => (
        <Tag
            data-test="saved-search-tag"
            style={{ cursor: 'pointer' }}
            key={search.name}
            value={search.name}
            onClick={boundClick}
        />
    );

    return (
        <TagList data-test="saved-search-tag-list" readOnly>
            {list.map(renderTag)}
        </TagList>
    );
}

SavedSearchTagList.propTypes = {
    list: PropTypes.array.isRequired,
    onSavedSearchClick: PropTypes.func.isRequired,
};

export default SavedSearchTagList;
