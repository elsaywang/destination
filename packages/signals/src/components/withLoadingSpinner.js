import React from 'react';
import Wait from '@react/react-spectrum/Wait';

function withLoadingSpinner(Component) {
    return function load({ isLoaded, ...props }) {
        return isLoaded ? <Component {...props} /> : <Wait />;
    };
}

export default withLoadingSpinner;
