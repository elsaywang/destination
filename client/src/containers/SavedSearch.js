import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Heading from '@react/react-spectrum/Heading';
import SavedSearchTagList from '../components/SavedSearchTagList';
import withLoadingSpinner from '../components/withLoadingSpinner';

class SavedSearch extends Component {
    componentDidMount() {
        this.props.getSavedSearch();
    }

    render() {
        const WrappedSavedSearchTagList = withLoadingSpinner(SavedSearchTagList);

        return (
            <div data-test="saved-search">
                <Heading size={6}>Saved Search</Heading>
                <WrappedSavedSearchTagList
                    isLoaded={Boolean(this.props.list.length)}
                    list={this.props.list}
                    onSavedSearchClick={this.props.onSavedSearchClick}
                    currentSearch={this.props.currentSearch}
                    deleteSearch={this.props.deleteSearch}
                />
            </div>
        );
    }
}

SavedSearch.propTypes = {
    list: PropTypes.array.isRequired,
    getSavedSearch: PropTypes.func.isRequired,
    onSavedSearchClick: PropTypes.func.isRequired,
    currentSearch: PropTypes.string.isRequired,
    deleteSearch: PropTypes.func.isRequired,
};

export default SavedSearch;
