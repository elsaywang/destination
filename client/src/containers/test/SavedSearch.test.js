import React from 'react';
import { shallow } from 'enzyme';
import Heading from '@react/react-spectrum/Heading';
import SavedSearch from '../SavedSearch';
import SavedSearchTagList from '../../components/SavedSearchTagList';
import withLoadingSpinner from '../../components/withLoadingSpinner';

describe('<SavedSearch /> component', () => {
    const mockFn = jest.fn();
    const list = [
        {
            id: 17966,
            name: 'Germaine Wiza',
            keyValuePairs: [
                {
                    key: 'k-monitor',
                    operator: '>',
                    value: 'v-programming',
                },
            ],
            type: 'Lek',
            source: 'Buckinghamshire',
            totalCount: 59925,
            percentageChange: 45886,
            includedTraits: [0, 1, 2],
            signalStatus: 'USED',
        },
    ];
    const currentSearch = {
        name: 'Watson Cartwright',
    };
    const wrapper = shallow(
        <SavedSearch
            list={list}
            getSavedSearch={mockFn}
            onSavedSearchClick={mockFn}
            currentSearch={currentSearch.name}
            deleteSearch={mockFn}
        />,
    );

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders saved search div', () => {
            expect(wrapper.find('[data-test="saved-search"]').exists()).toBe(true);
        });

        it('renders <Heading />', () => {
            expect(wrapper.find(Heading).exists()).toBe(true);
        });
    });
});
