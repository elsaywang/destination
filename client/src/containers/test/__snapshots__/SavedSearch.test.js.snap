// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedSearch /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedSearch
    currentSearch="Watson Cartwright"
    getSavedSearch={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    list={
      Array [
        Object {
          "id": 17966,
          "includedTraits": Array [
            0,
            1,
            2,
          ],
          "keyValuePairs": Array [
            Object {
              "key": "k-monitor",
              "operator": ">",
              "value": "v-programming",
            },
          ],
          "name": "Germaine Wiza",
          "percentageChange": 45886,
          "signalStatus": "USED",
          "source": "Buckinghamshire",
          "totalCounts": 59925,
          "type": "Lek",
        },
      ]
    }
    onSavedSearchClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Heading
          size={6}
        >
          Saved Search
        </Heading>,
        <load
          currentSearch="Watson Cartwright"
          isLoaded={true}
          list={
            Array [
              Object {
                "id": 17966,
                "includedTraits": Array [
                  0,
                  1,
                  2,
                ],
                "keyValuePairs": Array [
                  Object {
                    "key": "k-monitor",
                    "operator": ">",
                    "value": "v-programming",
                  },
                ],
                "name": "Germaine Wiza",
                "percentageChange": 45886,
                "signalStatus": "USED",
                "source": "Buckinghamshire",
                "totalCounts": 59925,
                "type": "Lek",
              },
            ]
          }
          onSavedSearchClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
        />,
      ],
      "data-test": "saved-search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Saved Search",
          "size": 6,
        },
        "ref": null,
        "rendered": "Saved Search",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentSearch": "Watson Cartwright",
          "isLoaded": true,
          "list": Array [
            Object {
              "id": 17966,
              "includedTraits": Array [
                0,
                1,
                2,
              ],
              "keyValuePairs": Array [
                Object {
                  "key": "k-monitor",
                  "operator": ">",
                  "value": "v-programming",
                },
              ],
              "name": "Germaine Wiza",
              "percentageChange": 45886,
              "signalStatus": "USED",
              "source": "Buckinghamshire",
              "totalCounts": 59925,
              "type": "Lek",
            },
          ],
          "onSavedSearchClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Heading
            size={6}
          >
            Saved Search
          </Heading>,
          <load
            currentSearch="Watson Cartwright"
            isLoaded={true}
            list={
              Array [
                Object {
                  "id": 17966,
                  "includedTraits": Array [
                    0,
                    1,
                    2,
                  ],
                  "keyValuePairs": Array [
                    Object {
                      "key": "k-monitor",
                      "operator": ">",
                      "value": "v-programming",
                    },
                  ],
                  "name": "Germaine Wiza",
                  "percentageChange": 45886,
                  "signalStatus": "USED",
                  "source": "Buckinghamshire",
                  "totalCounts": 59925,
                  "type": "Lek",
                },
              ]
            }
            onSavedSearchClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
          />,
        ],
        "data-test": "saved-search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Saved Search",
            "size": 6,
          },
          "ref": null,
          "rendered": "Saved Search",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentSearch": "Watson Cartwright",
            "isLoaded": true,
            "list": Array [
              Object {
                "id": 17966,
                "includedTraits": Array [
                  0,
                  1,
                  2,
                ],
                "keyValuePairs": Array [
                  Object {
                    "key": "k-monitor",
                    "operator": ">",
                    "value": "v-programming",
                  },
                ],
                "name": "Germaine Wiza",
                "percentageChange": 45886,
                "signalStatus": "USED",
                "source": "Buckinghamshire",
                "totalCounts": 59925,
                "type": "Lek",
              },
            ],
            "onSavedSearchClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
